commands/create_kubeconfig.go

```diff
@@ -452,43 +407,28 @@ func createKubeconfig(args createKubeconfigArguments) (createKubeconfigResult, e
 		}
 	} else {
 		// create a self-contained kubeconfig
-		kubeconfig := Kubeconfig{
-			APIVersion:     "v1",
-			Kind:           "Config",
-			CurrentContext: args.contextName,
-			Clusters: []KubeconfigNamedCluster{
-				{
-					Name: "giantswarm-" + args.clusterID,
-					Cluster: KubeconfigCluster{
-						Server:                   result.apiEndpoint,
-						CertificateAuthorityData: base64.StdEncoding.EncodeToString([]byte(response.Payload.CertificateAuthorityData)),
-					},
-				},
-			},
-			Contexts: []KubeconfigNamedContext{
-				{
-					Name: args.contextName,
-					Context: KubeconfigContext{
-						Cluster: "giantswarm-" + args.clusterID,
-						User:    "giantswarm-" + args.clusterID + "-user",
-					},
-				},
-			},
-			Users: []KubeconfigUser{
-				{
-					Name: "giantswarm-" + args.clusterID + "-user",
-					User: KubeconfigUserKeyPair{
-						ClientCertificateData: base64.StdEncoding.EncodeToString([]byte(response.Payload.ClientCertificateData)),
-						ClientKeyData:         base64.StdEncoding.EncodeToString([]byte(response.Payload.ClientKeyData)),
-					},
+		var yamlBytes []byte
+		{
+			c := k8srestconfig.Config{
+				Logger: microloggertest.New(),
+
+				Address:   result.apiEndpoint,
+				InCluster: false,
+				TLS: k8srestconfig.TLSClientConfig{
+					CAData:  []byte(response.Payload.CertificateAuthorityData),
+					CrtData: []byte(response.Payload.ClientCertificateData),
+					KeyData: []byte(response.Payload.ClientKeyData),
 				},
-			},
-		}
+			}
+			restConfig, err := k8srestconfig.New(c)
+			if err != nil {
+				return result, microerror.Mask(err)
+			}
 
-		// create YAML
-		yamlBytes, err := yaml.Marshal(&kubeconfig)
-		if err != nil {
-			return result, microerror.Mask(err)
+			yamlBytes, err = kubeconfig.NewKubeConfigForRESTConfig(context.Background(), restConfig, fmt.Sprintf("giantswarm-%s", args.clusterID), "")
```

Comment:
I don't like defining the context here. We should create it at the entrypoint to the command.

I think it should be here. https://github.com/giantswarm/gsctl/blob/786cb6418360a4d52dbb8ff57e18559a29795a0c/commands/create_kubeconfig.go#L163

Reply:
That's in `createKubeconfigPreRunOutput`, I don't think it's nice passing around the context which would use just only one time.

Reply:
The context should be created at the entry point and flow through all methods. Like here in `opsctl`. If each function call has its own context package it doesn't help.

https://github.com/giantswarm/opsctl/blob/a948553d66cd789673d6671d2e9635e6f20c7ea8/command/update/installation/command.go#L111

Reply:
@rossf7 I can create `context` in `createKubeconfigRunOutput` instead, WDYT?

Reply:
Yes creating it there is good. Thx!